src/182B30.c

```diff
@@ -184,9 +184,139 @@ void func_802549C0(void) {
+        decorationTable->rotationPivotOffsetY[i] = (s32)(actor->rotationPivotOffset.y * actor->scalingFactor);
+
+        decorationTable->rotX[i] = clamp_angle(actor->rotation.x) * 0.5f;
+        decorationTable->rotY[i] = clamp_angle(actor->rotation.y) * 0.5f;
+        decorationTable->rotZ[i] = clamp_angle(actor->rotation.z) * 0.5f;
+
+        i++;
+        if (i >= ARRAY_COUNT(decorationTable->posX)) {
+            i = 0;
+        }
+        decorationTable->unk_7D9 = i;
+    }
+}
+
+void func_80254C50(Actor* actor) {
+    Matrix4f sp18;
+    Matrix4f sp58;
+    Matrix4f sp98;
+    Matrix4f spD8;
+    Matrix4f sp118;
+    Matrix4f sp158;
+    Matrix4f sp198;
+    Matrix4f sp1D8;
+    Matrix4f sp218;
+    Matrix4f sp258;
+    s32 scale;
+    ActorPart* partTable;
+    DecorationTable* decorationTable;
+    f32 rotX, rotY, rotZ;
+    s32 temp_s0_2;
+    s32 temp_v1;
+    s32 temp_s0;
+    s32 temp_s1;
+    s32 phi_s6;
+    s32 phi_s4;
+    f32 x, y, z;
+
+    partTable = actor->partsTable;
+    decorationTable = partTable->decorationTable;
+    if (decorationTable->effectType != 0) {
+        decorationTable->effectType = decorationTable->effectType - 1;
+        if (decorationTable->effectType == 0) {
+            actor->flags &= ~ACTOR_FLAG_10000000;
+            return;
+        }
+    }
+
+    if (!(partTable->flags & ACTOR_PART_FLAG_INVISIBLE) && partTable->idleAnimations != NULL) {
+        s32 sp298 = 0;
+        s32 sp29C = 0;
+        s32 i = decorationTable->unk_7D9;
+
+        while (1) {
+            i--;
+            sp298 += 1;
+            if (i < 0) {
+                i = 0xF;
+            }
+            if (i == decorationTable->unk_7D9) {
+                break;
+            }
 
-INCLUDE_ASM(void, "182B30", func_80254C50);
+            if (sp298 >= 3) {
+                sp298 = 0;
+                sp29C++;
+
+                if (decorationTable->unk_7DA < sp29C) {
+                    break;
+                }
+                temp_v1 = partTable->opacity;
+                x = decorationTable->posX[i];
+                y = decorationTable->posY[i];
+                z = decorationTable->posZ[i];
+
+                scale = decorationTable->scale[i];
+
+                temp_s0 = decorationTable->rotationPivotOffsetX[i];
+                temp_s1 = decorationTable->rotationPivotOffsetY[i];
+
+                rotX = decorationTable->rotX[i] * 2;
+                rotY = decorationTable->rotY[i] * 2;
+                rotZ = decorationTable->rotZ[i] * 2;
+
+
+                phi_s6 = 0x78;
+                phi_s4 = 0x14;
+                if (temp_v1 < 0x32) {
+                    phi_s6 = 0x32;
+                    phi_s4 = 8;
+                } else if (temp_v1 < 0x64) {
+                    phi_s6 = 0x46;
+                    phi_s4 = 0xA;
+                } else if (temp_v1 < 0x96) {
+                    phi_s6 = 0x64;
+                    phi_s4 = 0xF;
+                }
```

Comment:
These should probably all be in base 10

Reply:
I agree, since they are part of the opacity calculation and co.